### ISSUE TITLE
Can't connect to host with space in name, with useLocalServer=false

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode-remote-release#5993]
While verifying, I realized that this only works for the local server case https://github.com/microsoft/vscode-remote-release/issues/5816

- Set `useLocalServer=false`
- Connect to an SSH host with a space in the Host name
- It fails while setting up the tunnel

**Comment by tanhakabir** [comment]
Are you sure? The only changes I made for the bug and what I tested locally was for the terminal runner only case. https://github.com/microsoft/vscode-remote-ssh/commit/feba7e792deb9852f4c46e68bc5ca3809d386009

**Comment by roblourens** [comment]
Yeah, to be clear, I mean connecting doesn't work. The clean command works fine

**Comment by tanhakabir** [comment]
Hm I'm still not reproing. This is what I'm doing:
![recording (59)](https://user-images.githubusercontent.com/12758612/145236412-62278278-4812-4305-ac70-777fbb9b05db.gif)

**Comment by tanhakabir** [comment]
@roblourens what I see in logs:
```
[09:23:40.956] Log Level: 2
[09:23:40.957] remote-ssh@0.66.1
[09:23:40.958] darwin x64
[09:23:40.958] SSH Resolver called for "ssh-remote+name with spaces", attempt 1
[09:23:40.959] "remote.SSH.useLocalServer": false
[09:23:40.959] "remote.SSH.showLoginTerminal": false
[09:23:40.959] "remote.SSH.remotePlatform": {"host":"linux","winhost":"windows","localhost":"macOS","name with spaces":"linux"}
[09:23:40.960] "remote.SSH.path": undefined
[09:23:40.960] "remote.SSH.configFile": 
[09:23:40.960] "remote.SSH.useFlock": true
[09:23:40.960] "remote.SSH.lockfilesInTmp": false
[09:23:40.960] "remote.SSH.localServerDownload": auto
[09:23:40.960] "remote.SSH.remoteServerListenOnSocket": false
[09:23:40.961] "remote.SSH.showLoginTerminal": false
[09:23:40.961] "remote.SSH.defaultExtensions": []
[09:23:40.961] "remote.SSH.loglevel": 2
[09:23:40.961] "remote.SSH.serverPickPortsFromRange": {"work-main":{"start":60000,"end":61000},"work-test":{"start":60000,"end":61000},"localhost":{"start":5000,"end":6000}}
[09:23:40.961] "remote.SSH.enableDynamicForwarding": true
[09:23:40.962] "remote.SSH.serverInstallPath": {"work-main":"/vscode_clean","work-test":"/vscode_sandbox"}
[09:23:40.963] SSH Resolver called for host: name with spaces
[09:23:40.963] Setting up SSH remote "name with spaces"
[09:23:40.974] Using commit id "7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e" and quality "insider" for server
[09:23:40.976] Install and start server if needed
[09:23:40.980] PATH: /usr/bin:/bin:/usr/sbin:/sbin
[09:23:40.980] Checking ssh with "ssh -V"
[09:23:41.028] > OpenSSH_8.6p1, LibreSSL 2.8.3

[09:23:41.032] Running script with connection command: ssh -T -D 65085 -o ConnectTimeout=15 'name with spaces' bash
[09:23:42.400] > root@147.182.250.143's password: 
[09:23:42.401] Got some output, clearing connection timeout
[09:23:42.401] Showing password prompt
[09:23:46.278] Got password response
[09:23:46.279] "install" wrote data to terminal: "************"
[09:23:46.296] > 
[09:23:47.177] > 8b8aa6a94001: running
> Acquiring lock on /root/.vscode-server-insiders/bin/7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e/vscode-remote-lock.root.7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e
> Found existing installation at /root/.vscode-server-insiders/bin/7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e...
> Checking /root/.vscode-server-insiders/.7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e.log and /root/.vscode-server-insiders/.7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e.pid for a running server
> Looking for server with pid: 545773
> Found running server...
>  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
> Checking server status on port 35953 with wget
> 8b8aa6a94001: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==35953==
> osReleaseId==ubuntu==
> arch==x86_64==
> tmpDir==/run/user/0==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> connectionToken==1111aa1a-1111-1111-1111-1a1aa1111aa1==
> 8b8aa6a94001: end
[09:23:47.178] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==35953==
osReleaseId==ubuntu==
arch==x86_64==
tmpDir==/run/user/0==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==1111aa1a-1111-1111-1111-1a1aa1111aa1==

[09:23:47.178] Remote server is listening on 35953
[09:23:47.178] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":35953},"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/0","platform":"linux","connectionToken":"1111aa1a-1111-1111-1111-1a1aa1111aa1"},"installUnpackCode":""}
[09:23:47.181] Starting forwarding server. localPort 65089 -> socksPort 65085 -> remotePort 35953
[09:23:47.181] Forwarding server listening on 65089
[09:23:47.182] Waiting for ssh tunnel to be ready
[09:23:47.182] Tunneled 35953 to local port 65089
[09:23:47.183] Resolved "ssh-remote+name with spaces" to "127.0.0.1:65089"
[09:23:47.183] [Forwarding server 65089] Got connection 0
[09:23:47.189] ------




[09:23:47.201] [Forwarding server 65089] Got connection 1
[09:23:47.342] [Forwarding server 65089] Got connection 2
```

**Comment by roblourens** [comment]
Oh, I disabled dynamic forwarding, and it doesn't work without that option
